src/btree/bt_cursor.c

```diff
@@ -481,19 +482,32 @@ __wt_btcur_search_uncommitted(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp)
      * itself. But, we cannot be sure of finding one, as pre processing of this prepared transaction
      * updates could have happened as part of resolving earlier transaction operations.
      */
-    if (cbt->compare != 0)
-        return (0);
+    if (cbt->compare == 0) {
+        /*
+         * Get the uncommitted update from the cursor. For column store there will be always a
+         * insert structure for updates irrespective of fixed length or variable length.
+         */
+        if (cbt->ins != NULL)
+            upd = cbt->ins->upd;
+        else if (cbt->btree->type == BTREE_ROW)
```

Comment:
Is there any reason to remove the assert and add the if conditions. Optionally consider the following alternative:
```
        if (cbt->ins != NULL)
            upd = cbt->ins->upd;
        else {
            WT_ASSERT(session, cbt->btree->type == BTREE_ROW && cbt->ref->page->modify != NULL &&
              cbt->ref->page->modify->mod_row_update != NULL);
            upd = cbt->ref->page->modify->mod_row_update[cbt->slot];
        }
```

Reply:
1449dd2

Thanks for picking that up. Reverted a few of these changes.

This was from before I knew of @kommiharibabu's change so it was accounting for when there's only a prepared update for a given key in lookaside but nothing in the update list. As I understand it, that's no longer possible.